### PR TITLE
JDK-8300651: Replace NULL with nullptr in share/runtime/

### PR DESCRIPTION
**Pull request by jdksjolen**
Do the conversion in the share/runtime/ sub-directory and all of its files.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300651](https://bugs.openjdk.org/browse/JDK-8300651): Replace NULL with nullptr in share/runtime/

### Reviewers
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**) ⚠️ Review applies to [64d2c224](https://git.openjdk.org/jdk/pull/12094/files/64d2c224316bf309bbf412ee0116657a62021092)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12094/head:pull/12094` \
`$ git checkout pull/12094`

Update a local copy of the PR: \
`$ git checkout pull/12094` \
`$ git pull https://git.openjdk.org/jdk pull/12094/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12094`

View PR using the GUI difftool: \
`$ git pr show -t 12094`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12094.diff">https://git.openjdk.org/jdk/pull/12094.diff</a>

</details>

### CODE REVIEW

**Comment by jdksjolen** [comment]
Seems like uploading the GHA results are failing.

**Review by robehn — APPROVED** [review]
Thank you!

File: src/hotspot/share/runtime/threadSMR.hpp [hunk 1]
```diff
@@ -347,7 +347,7 @@ class JavaThreadIterator : public StackObj {
 
 public:
   JavaThreadIterator(ThreadsList *list) : _list(list), _index(0) {
-    assert(list != NULL, "ThreadsList must not be NULL.");
+    assert(list != nullptr, "ThreadsList must not be nullptr.");
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/threadSMR.hpp:350]: null

File: src/hotspot/share/runtime/vmStructs.hpp [hunk 2]
```diff
@@ -106,27 +106,27 @@ typedef struct {
...
   // The last entry is identified over in the serviceability agent by
-  // the fact that it has a NULL fieldName
+  // the fact that it has a null fieldName
   static VMStructEntry localHotSpotVMStructs[];
   // The function to get localHotSpotVMStructs length
   static size_t localHotSpotVMStructsLength() NOT_VM_STRUCTS_RETURN_(0);
 
   // The last entry is identified over in the serviceability agent by
-  // the fact that it has a NULL typeName
+  // the fact that it has a nullptr typeName
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/vmStructs.hpp:115]: null for consistency

File: src/hotspot/share/runtime/mutexLocker.cpp [hunk 3]
```diff
@@ -172,7 +172,7 @@ static int _num_mutex;
 #ifdef ASSERT
 void assert_locked_or_safepoint(const Mutex* lock) {
   // check if this thread owns the lock (common case)
-  assert(lock != NULL, "Need non-NULL lock");
+  assert(lock != nullptr, "Need non-nullptr lock");
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/mutexLocker.cpp:175]: null for consistency

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Looks good but a few cases where nullptr should be null in comments for consistency with surrounding changes if nothing else.

Thanks.

**Comment by robehn** [comment]
> Looks good but a few cases where nullptr should be null in comments for consistency with surrounding changes if nothing else.

Yes, good. The terminology used by posix is: "null pointer", so using lower case null I think is good.

**Review by dholmes-ora — APPROVED** [review]
Looks good! Thanks.

**Comment by jdksjolen** [comment]
Passes tier1.

/integrate

**Pull request closed without merging**